### ISSUE TITLE
How to support a generic "has key" validator

### ISSUE THREAD

**Issue by cfjedimaster** [lirantal/swagger-lint-api#7]
**Is your feature request related to a problem? Please describe.**
Our particular implementation requires that the parameter values include an `x-required` field. (I'm not sure why they aren't using the generic `required` property.) 

I need to ensure that all objects under `parameters` include `x-required`.

**Describe the solution you'd like**
My validator will let me define what I require (`x-foo`) and under what it should look, so for example, `parameters`. The validator will find all instances of `parameters`, and will either:

* if an object, see if it has `x-foo`
* if an array, assume an array of objects and look for `x-foo`. 

Basically, I want to ensure X key always exists under Y (but support Y either being an object or array of objects).
